core/arch/aarch64/instr.c

```diff
@@ -53,6 +53,8 @@ instr_length_arch(dcontext_t *dcontext, instr_t *instr)
 {
     if (instr_get_opcode(instr) == OP_LABEL)
         return 0;
+    if (instr_get_opcode(instr) == OP_ldstex)
+        return opnd_get_immed_int(instr_get_src(instr, 0)) * AARCH64_INSTR_SIZE;
```

Comment:
?  Someone clearing the length field would break x86 so I don't think that's the case.